### PR TITLE
Update to use terra-doc-template

### PR DESCRIPTION
**Pull request by emilyrohrbough**
### Summary
Update site pages to use terra-doc-template.

### CODE REVIEW

**Comment by bjankord** [comment]
With this change, I'd like to delete most of the usage sections from the README file. The ability to view the example source code should mean we don't need the usage examples anymore / we no longer need to maintain them.

However, I think there are cases where examples in the README are helpful, for instance, the examples in the app delegate package. 

I'd use this guidance, if the toggleable source code matches or nearly matches the usage example in the README file, delete the usage example from the README.

**Comment by emilyrohrbough** [comment]
I can go through and look for duplicate examples, however are there any off the top that you know you want gone?

File: packages/terra-popup/examples/Index.site-page.jsx [hunk 1]
```diff
@@ -19,83 +14,59 @@ import PopupNoHeader from './site-examples/PopupNoHeader';
...
-  </div>
+const DocPage = () => (
+  <DocTemplate
+    packageName={name}
+    readme={ReadMe}
+    srcPath={`https://github.com/cerner/terra-framework/tree/master/packages/${name}`}
+    examples={[
+      {
+        title: 'Popup Default Props',
+        example: <PopupStandard />,
```
Comment (StephenEsser) [inline packages/terra-popup/examples/Index.site-page.jsx]: Should these have a source? Or should the doc template conditionally render the example?

image

Comment (emilyrohrbough) [reply packages/terra-popup/examples/Index.site-page.jsx]: Both. I wasn't intending to add src for each popup example- updating now but the template should also be conditional IMO

**Pull request merged**